driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -166,14 +189,14 @@ impl UpdatingOrderbook {
 }
 
 impl StableXOrderBookReading for UpdatingOrderbook {
-    /// Blocks on updating the orderbook. When this is called the first time this can take several
-    /// minutes.
+    /// Blocks on updating the orderbook. This can be expensive if `initialize` hasn't been called before.
     fn get_auction_data(&self, batch_id_to_solve: U256) -> Result<(AccountState, Vec<Order>)> {
-        let mut context = self
-            .context
-            .lock()
-            .map_err(|err| anyhow!("poison error: {}", err))?;
-        self.update_with_events(&mut context).wait()?;
-        context.orderbook.get_auction_data(batch_id_to_solve)
+        self.do_with_context(|context| {
+            self.update_with_events(context).wait()?;
```

Comment:
Now, if the optional `initialize` is not called, we `update_with_events` twice in a row (fetching at least 25 blocks). Not a big issue, but why don't we do the update_with_events unconditionally but only once inside `do_with_context` and maybe call it `do_with_updated_context` ?

Reply:
I didn't think of that when I wrote this code but after trying your idea out I realized that the side effect of the  current version that the orderbook is updated twice is actually a positive. The initial update with all past events can take a long time so it makes sense to do the second small update again in case there have been new blocks since then.
I'll add a comment to explain this and keep the behavior.